sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/PassThroughThenCleanup.java

```diff
@@ -79,6 +81,6 @@ public void processElement(ProcessContext c) {
   }
 
   abstract static class CleanupOperation implements Serializable {
-    abstract void cleanup(PipelineOptions options) throws Exception;
+    abstract void cleanup(DoFn.ProcessContext options) throws Exception;
```

Comment:
Should probably not take a raw type, nor should it take a DoFn.ProcessContext in general, because it shouldn't be allowed to do anything with the context except get the pipeline options and have access to a side input. Consider instead mimicking (or extracting) the pattern from https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinations.java#L76 - which conveniently enough is also in BigQuery.